react-frontend/src/api.js

```diff
@@ -27,9 +37,42 @@ export const login = async (email, password) => {
 
 export const getListings = async () => {
   try {
-    const response = await axios.get(`${API_BASE_URL}/listings`)
+    const response = await axiosInstance.get('/listings')
+    return response.data
+  } catch (error) {
+    throw error.response.data
+  }
+}
+
+export const createListing = async (listingData) => {
+  try {
+    const response = await axiosInstance.post('/listings', listingData)
+    return response.data
+  } catch (error) {
+    throw error.response.data
+  }
+}
+
+export const getListingById = async (listingId) => {
+  const response = await axios.get(`${API_BASE_URL}/listings/${listingId}`)
+  return response.data
+}
+
+export const getBidsByListingId = async (listingId) => {
+  const response = await axios.get(`${API_BASE_URL}/bids/${listingId}`)
+  return response.data
+}
+
+export const placeBid = async (listingId, amount) => {
+  try {
+    const response = await axiosInstance.post(`/bids/${listingId}/place`, { amount })
     return response.data
   } catch (error) {
     throw error.response.data
```

Comment:
nit:It's better to throw an Error. Like `throw Error(error.response.data)`

Reply:
Agree, refactored in [9667654](https://github.com/YaroslavKSE/AuctionWeb/pull/2/commits/9667654d41d44244545551d6fbdb5bedb5025c79)